src/ussd_pmtct_rapidpro.js

```diff
@@ -1,24 +1,413 @@
             return new EndState(name, {
-                text: $(
-                    "You need to be registered on MomConnect to receive these messages. Please visit the nearest " +
-                    "clinic to register."
+                next: "state_start",
+                text:
+                    "Thanks! MomConnect will continue to send HER helpful messages and process your " +
+                    "personal info."
+            });
+        });
+
+        self.add("state_optout_reason", function(name) {
+            var question = "Please tell us why she no longer wants to get msgs:";
+            return new PaginatedChoiceState(name, {
+                question: question,
+                error: question,
+                accept_labels: true,
+                options_per_page: null,
+                characters_per_page: 160,
+                choices: [
+                    new Choice("not_hiv_positive", "She's not HIV+"),
+                    new Choice("miscarriage", "Miscarriage"),
+                    new Choice("stillborn", "Baby was stillborn"),
+                    new Choice("babyloss", "Baby passed away"),
+                    new Choice("not_useful", "Msgs aren't helpful"),
+                    new Choice("other", "Other"),
+                    new Choice("unknown", "I prefer not to say")
+                ],
+                next: function(choice) {
+                    self.im.user.set_answer("optout_reason", choice.value);
+                    if(_.includes(["miscarriage", "stillborn", "babyloss"], choice.value)) {
+                        return "state_loss_optout";
+                    } else {
+                        return "state_opted_out";
+                      }  
+                }
+            });
+        });
+
+        self.states.add("state_opted_out", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text:
+                    "Thank you. She will no longer receive messages from us about HIV. " +
+                    "For any medical concerns, please visit a clinic."
+            });
+        });
+
+        self.add("state_loss_optout", function(name) {
+            return new ChoiceState(name, {
+                question: "We're sorry for your loss. Would she like to receive a small set of " +
+                            "MomConnect messages that could help you during this difficult time?",
+                error:
+                    "Sorry, please reply with the number next to your answer. " +
+                    "Would she like to receive a small set of MomConnect messages?",
+                choices: [
+                    new Choice("yes", "Yes"),
+                    new Choice("no", "No")
+                ],
+                next: function(choice) {
+                    if(choice.value === "yes") {
+                        return "state_loss_subscription";
+                    } else {
+                        return "state_opted_out";
```

Comment:
We need to be submitting the optout to RapidPro here, not going to the end state

Reply:
same as above